apollo-router/src/protocols/sse/client.rs

```diff
@@ -0,0 +1,638 @@
+                } => match uri_from_header(maybe_header) {
+                    Ok(uri) => {
+                        *self.as_mut().project().current_url = uri;
+                        self.as_mut().project().state.set(State::New);
+                    }
+                    Err(e) => {
+                        self.as_mut().project().state.set(State::StreamClosed);
+                        return Poll::Ready(Some(Err(e)));
+                    }
+                },
+                StateProj::Connected { body } => match ready!(body.poll_data(cx)) {
+                    Some(Ok(result)) => {
+                        this.event_parser.process_bytes(result)?;
+                        continue;
+                    }
+                    Some(Err(e)) => {
+                        if self.props.reconnect_opts.reconnect {
+                            let duration = self
+                                .as_mut()
+                                .project()
+                                .retry_strategy
+                                .next_delay(Instant::now());
+                            self.as_mut()
+                                .project()
+                                .state
+                                .set(State::WaitingToReconnect {
+                                    sleep: delay(duration, "reconnecting"),
+                                });
+                        }
+
+                        if let Some(cause) = e.source() {
+                            if let Some(downcast) = cause.downcast_ref::<std::io::Error>() {
+                                if let std::io::ErrorKind::TimedOut = downcast.kind() {
+                                    return Poll::Ready(Some(Err(Error::TimedOut)));
+                                }
+                            }
+                        } else {
+                            return Poll::Ready(Some(Err(Error::HttpStream(Box::new(e)))));
+                        }
+                    }
+                    None => {
+                        let duration = self
+                            .as_mut()
+                            .project()
+                            .retry_strategy
+                            .next_delay(Instant::now());
+                        self.as_mut()
+                            .project()
+                            .state
+                            .set(State::WaitingToReconnect {
+                                sleep: delay(duration, "retrying"),
+                            });
+
+                        if self.event_parser.was_processing() {
+                            return Poll::Ready(Some(Err(Error::UnexpectedEof)));
+                        }
+                        return Poll::Ready(Some(Err(Error::Eof)));
+                    }
+                },
+                StateProj::WaitingToReconnect { sleep: delay } => {
+                    ready!(delay.poll(cx));
+                    tracing::info!("Reconnecting");
```

Comment:
```suggestion
                    tracing::info!("Reconnecting sse connection", url = ?self.current_url);
```
I don't know if it compiles but you got the main idea